packages/labs/src/components/tag-input/_tag-input.scss

```diff
@@ -45,6 +45,11 @@ $ti-icon-padding-large: ($pt-input-height-large - $pt-icon-size-standard) / 2;
     padding: 0 ($input-padding-horizontal - $ti-padding);
     line-height: $tag-height;
 
+    // match padding of other input elements iff no tags are selected
+    &:first-child {
```

Comment:
Scratch that, it looks great, with or without tags/left icon:
![image](https://user-images.githubusercontent.com/199754/29628144-78ed9db0-87e9-11e7-89cb-9bbbe8515d68.png)
![image](https://user-images.githubusercontent.com/199754/29628159-87649c36-87e9-11e7-9f20-409ccd9b9ea7.png)